pkg/startup/v7/derive.go

```diff
@@ -135,3 +139,216 @@ func (derivedType) CaBundle(cs *api.OpenShiftManagedCluster) ([]*x509.Certificat
+            <Column name="ScanName" mdstype="mt:wstr" type="str"></Column>
+            <Column name="EventTime" mdstype="mt:utc" type="str-rfc3339"></Column>
+            <Column name="Data" mdstype="mt:wstr" type="str"></Column>
+            <Column name="PartCount" mdstype="mt:int32" type="str"></Column>
+            <Column name="PartIndex" mdstype="mt:int32" type="str"></Column>
+        </Schema>
+    </Schemas>
+    <Sources>
+        <Source name="heartbeat" schema="azsecv1"/>
+        <Source name="baseline" schema="azsecv1"/>
+        <Source name="software" schema="azsecv1"/>
+        <Source name="scan_event" schema="azsecv2"/>
+        <Source name="alert" schema="azsecv2"/>
+        <Source name="audit" dynamic_schema="true"/>
+        <Source name="journald" dynamic_schema="true"/>
+        <Source name="k8s_audit" dynamic_schema="true"/>
+    </Sources>
+    <Events>
+        <MdsdEvents>
+            <MdsdEventSource source="heartbeat">
+                <RouteEvent eventName="LinuxAsmHeartbeat" priority="Normal" storeType="CentralBond"></RouteEvent>
+            </MdsdEventSource>
+            <MdsdEventSource source="baseline">
+                <RouteEvent eventName="LinuxAsmBaseline" priority="Normal" storeType="CentralBond"></RouteEvent>
+            </MdsdEventSource>
+            <MdsdEventSource source="software">
+                <RouteEvent eventName="LinuxAsmSoftware" priority="Normal" storeType="CentralBond"></RouteEvent>
+            </MdsdEventSource>
+			<MdsdEventSource source="scan_event">
+                <RouteEvent eventName="LinuxAsmScanEvent" priority="Normal" storeType="CentralBond"></RouteEvent>
+            </MdsdEventSource>
+			<MdsdEventSource source="alert">
+                <RouteEvent eventName="LinuxAsmAlert" priority="Normal" storeType="CentralBond"></RouteEvent>
+            </MdsdEventSource>
+            <MdsdEventSource source="audit">
+                <RouteEvent eventName="LinuxAsmAudit" priority="Normal" storeType="CentralBond"></RouteEvent>
+            </MdsdEventSource>
+            <MdsdEventSource source="journald">
+                <RouteEvent eventName="CustomerSyslogEvents" storeType="CentralBond" priority="High"/>
+            </MdsdEventSource>
+            <MdsdEventSource source="k8s_audit">
+                <RouteEvent eventName="CustomerAuditLogEvents" storeType="CentralBond" priority="High"/>
+            </MdsdEventSource>
+        </MdsdEvents>
+    </Events>
+</MonitoringManagement>`
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return "", err
+	}
+	role := names.GetAgentRole(hostname)
+
+	t := template.Must(template.New("").Funcs(map[string]interface{}{
+		"XMLEscape": func(s string) (string, error) {
+			var b bytes.Buffer
+			err := xml.EscapeText(&b, []byte(s))
+			return b.String(), err
+		},
+	}).Parse(tmpl))
+
+	b := &bytes.Buffer{}
+
+	err = t.Execute(b, map[string]string{
+		"Namespace":         cs.Config.GenevaLoggingNamespace,
```

Comment:
I'd rather not encourage using intermediate variable names (I know this is a moved function, but this PR is then building on the pattern).  Best just to pass in cs, then refer to `{{ .ContainerService.Foo }}` like we do everywhere else.  Try to get to the point where template.Template() is being used and (reluctantly) add the XMLEscape function there.